### PR TITLE
fix(manager/github-actions): parse dependencies with double quotes

### LINKED ISSUE #10033: GitHub Actions config file is skipped if there are actions wrapped with quotations in the file
**What Renovate type, platform and version are you using?**

**Renovate type**

- [X] Hosted app
- [ ] Self hosted

**Renovate platform**

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server (self-hosted)
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server (self-hosted)
- [ ] Gitea (gitea.com, self-hosted)
- [X] GitHub (github.com, GitHub Enterprise)
- [ ] GitLab (gitlab.com, self-hosted)

**Renovate version (if self hosting)**

N/A

**Describe the bug**

For GitHub Actions config files, if there are actions wrapped with quotations in the file, renovate does not check for actions updates for that file. See [this](https://github.com/zeshuaro/telegram-pdf-bot/blob/323181a64d419c1aaaa22969923aa65804cde813/.github/workflows/auto-merge.yml#L27) file for example. Related discussion #10031

Note that renovate was able to check for updates in the config file after removing the quotations, see [here](https://github.com/zeshuaro/telegram-pdf-bot/blob/3a19829fc4a38fa72244af218d1f2377bd80e93a/.github/workflows/auto-merge.yml) for the updated file.

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
DEBUG: Matched 4 file(s) for manager github-actions: .github/workflows/auto-merge.yml, .github/workflows/codeql-analysis.yml, .github/workflows/update-l10n-sources.yml, .github/workflows/update-l10n-translations.yml

DEBUG: packageFiles with updates
{
  "config": {
    "dockerfile": [
      {
        "packageFile": "Dockerfile",
        "deps": [
          {
            "depName": "python",
            "currentValue": "3.9.5",
            "replaceString": "python:3.9.5",
            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "datasource": "docker",
            "depType": "final",
            "depIndex": 0,
            "updates": [],
            "warnings": [],
            "currentVersion": "3.9.5",
            "fixedVersion": "3.9.5"
          }
        ]
      }
    ],
    "github-actions": [
      {
        "packageFile": ".github/workflows/codeql-analysis.yml",
        "deps": [
          {
            "depName": "actions/checkout",
            "currentValue": "v2",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "pinDigests": false,
            "depIndex": 0,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/actions/checkout",
            "currentVersion": "v2",
            "fixedVersion": "v2"
          },
          {
            "depName": "github/codeql-action",
            "currentValue": "v1",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "pinDigests": false,
            "depIndex": 1,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/github/codeql-action"
          },
          {
            "depName": "github/codeql-action",
            "currentValue": "v1",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "pinDigests": false,
            "depIndex": 2,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/github/codeql-action"
          },
          {
            "depName": "github/codeql-action",
            "currentValue": "v1",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "pinDigests": false,
            "depIndex": 3,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/github/codeql-action"
          }
        ]
      },
      {
        "packageFile": ".github/workflows/update-l10n-sources.yml",
        "deps": [
          {
            "depName": "actions/checkout",
            "currentValue": "v2",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "pinDigests": false,
            "depIndex": 0,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/actions/checkout",
            "currentVersion": "v2",
            "fixedVersion": "v2"
          },
          {
            "depName": "actions/setup-python",
            "currentValue": "v2",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "pinDigests": false,
            "depIndex": 1,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/actions/setup-python",
            "currentVersion": "v2",
            "fixedVersion": "v2"
          },
          {
            "depName": "actions/cache",
            "currentValue": "v2",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "pinDigests": false,
            "depIndex": 2,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/actions/cache",
            "currentVersion": "v2",
            "fixedVersion": "v2"
          },
          {
            "depName": "stefanzweifel/git-auto-commit-action",
            "currentValue": "v4",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "pinDigests": false,
            "depIndex": 3,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/stefanzweifel/git-auto-commit-action",
            "currentVersion": "v4",
            "fixedVersion": "v4"
          },
          {
            "depName": "crowdin/github-action",
            "currentValue": "1.1.0",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "pinDigests": false,
            "depIndex": 4,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/crowdin/github-action",
            "currentVersion": "1.1.0",
            "fixedVersion": "1.1.0"
          }
        ]
      },
      {
        "packageFile": ".github/workflows/update-l10n-translations.yml",
        "deps": [
          {
            "depName": "actions/checkout",
            "currentValue": "v2",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "pinDigests": false,
            "depIndex": 0,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/actions/checkout",
            "currentVersion": "v2",
            "fixedVersion": "v2"
          },
          {
            "depName": "crowdin/github-action",
            "currentValue": "1.1.0",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "pinDigests": false,
            "depIndex": 1,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/crowdin/github-action",
            "currentVersion": "1.1.0",
            "fixedVersion": "1.1.0"
          }
        ]
      }
    ],
    "pip_requirements": [
      {
        "packageFile": "requirements.txt",
        "deps": [
          {
            "depName": "Babel",
            "currentValue": "==2.9.1",
            "datasource": "pypi",
            "currentVersion": "2.9.1",
            "depIndex": 0,
            "updates": [],
            "warnings": [],
            "homepage": "http://babel.pocoo.org/",
            "fixedVersion": "2.9.1"
          },
          {
            "depName": "google-cloud-datastore",
            "currentValue": "==2.1.2",
            "datasource": "pypi",
            "currentVersion": "2.1.2",
            "depIndex": 1,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/googleapis/python-datastore",
            "fixedVersion": "2.1.2"
          },
          {
            "depName": "humanize",
            "currentValue": "==3.5.0",
            "datasource": "pypi",
            "currentVersion": "3.5.0",
            "depIndex": 2,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/jmoiron/humanize",
            "changelogUrl": "https://github.com/jmoiron/humanize/releases",
            "fixedVersion": "3.5.0"
          },
          {
            "depName": "img2pdf",
            "currentValue": "==0.4.1",
            "datasource": "pypi",
            "currentVersion": "0.4.1",
            "depIndex": 3,
            "updates": [],
            "warnings": [],
            "homepage": "https://gitlab.mister-muffin.de/josch/img2pdf",
            "fixedVersion": "0.4.1"
          },
          {
            "depName": "Logbook",
            "currentValue": "==1.5.3",
            "datasource": "pypi",
            "currentVersion": "1.5.3",
            "depIndex": 4,
            "updates": [],
            "warnings": [],
            "homepage": "http://logbook.pocoo.org/",
            "fixedVersion": "1.5.3"
          },
          {
            "depName": "matplotlib",
            "currentValue": "==3.4.2",
            "datasource": "pypi",
            "currentVersion": "3.4.2",
            "depIndex": 5,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/matplotlib/matplotlib",
            "homepage": "https://matplotlib.org",
            "fixedVersion": "3.4.2"
          },
          {
            "depName": "ocrmypdf",
            "currentValue": "==12.0.1",
            "datasource": "pypi",
            "currentVersion": "12.0.1",
            "depIndex": 6,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/jbarlow83/OCRmyPDF",
            "fixedVersion": "12.0.1"
          },
          {
            "depName": "pdfCropMargins",
            "currentValue": "==1.0.5",
            "datasource": "pypi",
            "currentVersion": "1.0.5",
            "depIndex": 7,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/abarker/pdfCropMargins",
            "fixedVersion": "1.0.5"
          },
          {
            "depName": "pdf2image",
            "currentValue": "==1.15.1",
            "datasource": "pypi",
            "currentVersion": "1.15.1",
            "depIndex": 8,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/Belval/pdf2image",
            "fixedVersion": "1.15.1"
          },
          {
            "depName": "pdfminer.six",
            "currentValue": "==20201018",
            "datasource": "pypi",
            "currentVersion": "20201018",
            "depIndex": 9,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/pdfminer/pdfminer.six",
            "fixedVersion": "20201018"
          },
          {
            "depName": "python-dotenv",
            "currentValue": "==0.17.1",
            "datasource": "pypi",
            "currentVersion": "0.17.1",
            "depIndex": 10,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/theskumar/python-dotenv",
            "fixedVersion": "0.17.1"
          },
          {
            "depName": "python-telegram-bot",
            "currentValue": "==13.5",
            "datasource": "pypi",
            "currentVersion": "13.5",
            "depIndex": 11,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/python-telegram-bot/python-telegram-bot",
            "homepage": "https://python-telegram-bot.org/",
            "changelogUrl": "https://python-telegram-bot.readthedocs.io/en/stable/changelog.html",
            "fixedVersion": "13.5"
          },
          {
            "depName": "Pillow",
            "currentValue": "==8.2.0",
            "datasource": "pypi",
            "currentVersion": "8.2.0",
            "depIndex": 12,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/python-pillow/Pillow",
            "homepage": "https://python-pillow.org",
            "changelogUrl": "https://github.com/python-pillow/Pillow/blob/master/CHANGES.rst",
            "fixedVersion": "8.2.0"
          },
          {
            "depName": "PyPDF2",
            "currentValue": "==1.26.0",
            "datasource": "pypi",
            "currentVersion": "1.26.0",
            "depIndex": 13,
            "updates": [],
            "warnings": [],
            "sourceUrl": "http://mstamy2.github.com/PyPDF2",
            "fixedVersion": "1.26.0"
          },
          {
            "depName": "slackclient",
            "currentValue": "==2.9.3",
            "datasource": "pypi",
            "currentVersion": "2.9.3",
            "depIndex": 14,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/slackapi/python-slackclient",
            "fixedVersion": "2.9.3"
          },
          {
            "depName": "textblob",
            "currentValue": "==0.15.3",
            "datasource": "pypi",
            "currentVersion": "0.15.3",
            "depIndex": 15,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/sloria/TextBlob",
            "fixedVersion": "0.15.3"
          },
          {
            "depName": "WeasyPrint",
            "currentValue": "==52.5",
            "datasource": "pypi",
            "currentVersion": "52.5",
            "depIndex": 16,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/Kozea/WeasyPrint",
            "homepage": "https://courtbouillon.org/weasyprint",
            "fixedVersion": "52.5"
          }
        ]
      }
    ]
  }
}
```

</details>

**Have you created a minimal reproduction repository?**

See [here](https://github.com/zeshuaro/telegram-pdf-bot/tree/323181a64d419c1aaaa22969923aa65804cde813) for the repo where renovate failed to check for [this](https://github.com/zeshuaro/telegram-pdf-bot/blob/323181a64d419c1aaaa22969923aa65804cde813/.github/workflows/auto-merge.yml) config file.

- [ ] I have provided a minimal reproduction repository
- [X] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

@rarkins suspected that renovate's parsing is getting tricked by the quotation marks in the config file

### PR DESCRIPTION
**Pull request by StinkyLord**
## Changes:

Fixed an issue where parsing GitHub workflow files is ignoring dependencies with double quotes

## Context:

When scanning github-actions some yml files have text in between double quotes, 
the code is using a regex to get these dependency text lines, but failed to manage with extra double quotes on both sides.

- fixes #10033

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/github-actions/extract.spec.ts [hunk 1]
```diff
@@ -18,5 +18,12 @@ describe('manager/github-actions/extract', () => {
         res.deps.filter((d) => d.datasource === 'github-tags')
       ).toHaveLength(5);
     });
+    it('extracts tag line with double quotes', () => {
+      const res = extractPackageFile(Fixtures.get('workflow_3.yml'));
+      expect(res.deps).toMatchSnapshot();
+      expect(
+        res.deps.filter((d) => d.datasource === 'github-tags')
+      ).toHaveLength(1);
```
Comment (viceice) [inline lib/manager/github-actions/extract.spec.ts:26]:
```suggestion
      expect(res.deps).toMatchSnapshot([
  {
    "currentValue": "v0.13.1",
    "datasource": "github-tags",
    "depName": "pascalgn/automerge-action",
    "depType": "action",
    "replaceString": "pascalgn/automerge-action@v0.13.1",
    "versioning": "docker",
  },
]);
```

some more explict matches

File: lib/manager/github-actions/extract.ts [hunk 2]
```diff
@@ -16,10 +16,11 @@ const shaRe = regEx(/^[a-z0-9]{40}|[a-z0-9]{64}$/);
 export function extractPackageFile(content: string): PackageFile | null {
   logger.trace('github-actions.extractPackageFile()');
   const deps: PackageDependency[] = [];
-  for (const line of content.split(newlineRegex)) {
+  for (let line of content.split(newlineRegex)) {
     if (line.trim().startsWith('#')) {
       continue;
     }
+    line = line.replaceAll('"', '');
```
Comment (viceice) [inline lib/manager/github-actions/extract.ts:23]: This should be done after dockerMatch

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/manager/github-actions/extract.ts [hunk 3]
```diff
@@ -31,6 +31,7 @@ export function extractPackageFile(content: string): PackageFile | null {
       continue;
     }
 
+    line = line.replaceAll('"', '');
```
Comment (rarkins) [inline lib/manager/github-actions/extract.ts:34]: According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll, `replaceAll` was only introduced in Node 15, while Renovate supports Node 14 still: https://github.com/renovatebot/renovate/blob/f0a1b64b15963114fd3976f8a77e4ff4134748c6/package.json#L123

Also, are you sure that `"` wouldn't be used within any strings validly?

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/manager/github-actions/extract.ts (hunk 3 above)
Comment (StinkyLord) [reply lib/manager/github-actions/extract.ts:34]: well i didnt know that, thanx , 
so its better to add the check into the regex as far as i can see.
ill add  `\"?`  to the regex on both sides

Comment (rarkins) [reply lib/manager/github-actions/extract.ts:34]: Trim it and remove any leading and trailing quotation marks?

Comment (rarkins) [reply lib/manager/github-actions/extract.ts:34]: Assuming you can isolate the value

File: lib/manager/github-actions/extract.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ import type { PackageDependency, PackageFile } from '../types';
 
 const dockerRe = regEx(/^\s+uses: docker:\/\/([^"]+)\s*$/);
 const actionRe = regEx(
-  /^\s+-?\s+?uses: (?<replaceString>(?<depName>[\w-]+\/[\w-]+)(?<path>\/.*)?@(?<currentValue>.+?)(?: # renovate: tag=(?<tag>.+?))?)\s*?$/
+  /^\s+-?\s+?uses: "?(?<replaceString>(?<depName>[\w-]+\/[\w-]+)(?<path>\/.*)?@(?<currentValue>.+?)(?: # renovate: tag=(?<tag>.+?))?)"?\s*?$/
```
Comment (rarkins) [inline lib/manager/github-actions/extract.ts:10]:
```suggestion
  /^\s+-?\s+?uses: "?(?<replaceString>(?<depName>[\w-]+\/[\w-]+)(?<path>\/.*)?@(?<currentValue>.+?)"?(?: # renovate: tag=(?<tag>.+?))?)\s*?$/
```

I don't think you want to wrap the comment?

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/manager/github-actions/extract.ts (hunk 4 above)
Comment (viceice) [reply lib/manager/github-actions/extract.ts:10]: This should also support single quotes

**Review by rarkins — DISMISSED** [review]

**Comment by rarkins** [comment]
Needs conflict resolving

**Review by rarkins — CHANGES_REQUESTED** [review]
I think there's been a bad commit merge. This PR now changes 170 files

**Review by rarkins — CHANGES_REQUESTED** [review]
I think there's been a bad commit merge. This PR now changes 170 files

**Review by rarkins — CHANGES_REQUESTED** [review]
I think there's been a bad commit merge. This PR now changes 170 files

**Review by rarkins — CHANGES_REQUESTED** [review]
I think there's been a bad commit merge. This PR now changes 170 files

**Comment by rarkins** [comment]
I think there's been a bad commit merge. This PR now changes 170 files

**Comment by StinkyLord** [comment]
ill re-open a new PR i merged manually and pushed everything back

**Comment by StinkyLord** [comment]
opened a new pull request : https://github.com/renovatebot/renovate/pull/14130

**Pull request closed without merging**